packages/suite/src/actions/wallet/stakeActions.ts

```diff
@@ -0,0 +1,213 @@
+    // if transaction is not signed yet interrupt signing in TrezorConnect
+    if (!signedTx) {
+        TrezorConnect.cancel('tx-cancelled');
+        return;
+    }
+    // otherwise just close modal
+    dispatch(modalActions.onCancel());
+};
+
+// private, called from signTransaction only
+const pushTransaction =
+    (signedTransaction: SignedTransaction['signedTransaction']) =>
+    async (dispatch: Dispatch, getState: GetState) => {
+        const { signedTx, precomposedTx } = getState().wallet.stake;
+        const { account } = getState().wallet.selectedAccount;
+        const device = selectDevice(getState());
+        if (!signedTx || !precomposedTx || !account) return;
+
+        const sentTx = await TrezorConnect.pushTransaction(signedTx);
+        // const sentTx = { success: true, payload: { txid: 'ABC ' } };
+
+        // close modal regardless result
+        dispatch(modalActions.onCancel());
+
+        const { token } = precomposedTx;
+        const spentWithoutFee = !token
+            ? new BigNumber(precomposedTx.totalSpent).minus(precomposedTx.fee).toString()
+            : '0';
+
+        // get total amount without fee OR token amount
+        const formattedAmount = token
+            ? `${formatAmount(
+                  precomposedTx.totalSpent,
+                  token.decimals,
+              )} ${token.symbol!.toUpperCase()}`
+            : formatNetworkAmount(spentWithoutFee, account.symbol, true, false);
+
+        if (sentTx.success) {
+            const { txid } = sentTx.payload;
+            dispatch(
+                notificationsActions.addToast({
+                    type: 'tx-sent',
+                    formattedAmount,
+                    device,
+                    descriptor: account.descriptor,
+                    symbol: account.symbol,
+                    txid,
+                }),
+            );
+
+            if (precomposedTx.prevTxid) {
+                // notification from the backend may be delayed.
+                // modify affected transaction(s) in the reducer until the real account update occurs.
+                // this will update transaction details (like time, fee etc.)
+                dispatch(
+                    replaceTransactionThunk({
+                        precomposedTx,
+                        newTxid: txid,
+                        signedTransaction,
+                    }),
+                );
+            }
+
+            // notification from the backend may be delayed.
+            // modify affected account balance.
+            // TODO: make it work with ETH accounts
+
+            // there is no point in fetching account data right after tx submit
+            //  as the account will update only after the tx is confirmed
+            dispatch(syncAccountsWithBlockchainThunk(account.symbol));
+
+            // handle metadata (labeling) from send form
+            const { metadata } = getState();
+            if (metadata.enabled) {
```

Comment:
labeling logic is not related to staking

Reply:
https://github.com/everstake/trezor-suite/pull/4/commits/2ed39aaff08c6509b0f48d27cef85971e157b43d